lib/bpf/bpf.go

```diff
@@ -233,9 +233,15 @@ func (s *Service) OpenSession(ctx *SessionContext) (uint64, error) {
 		return 0, trace.Wrap(err)
 	}
 
-	// Register cgroup in the BPF module.
-	if err := s.open.startSession(cgroupID); err != nil {
-		return 0, trace.Wrap(err)
+	for _, module := range []interface{ startSession(cgroupID uint64) error }{
+		s.open,
+		s.exec,
+		s.conn,
+	} {
+		// Register cgroup in the BPF module.
+		if err := module.startSession(cgroupID); err != nil {
+			return 0, trace.Wrap(err)
+		}
```

Comment:
not very familiar with bpf. but if say the second or the third in the loop fails, do we have to cleanup the earlier ones?

Reply:
Good point